src/apdu_sign.c

```diff
@@ -484,26 +477,30 @@ void parse_operation_inner(struct maybe_transaction* _U_ dest, bip32_path_t* _U_
 	}
 }
 
-int set_current_lock_arg(key_pair_t *key_pair) {
+int set_lock_arg(key_pair_t *key_pair, uint8_t *destination) {
+  check_null(key_pair);
+  check_null(destination);
   cx_blake2b_t lock_arg_state;
   cx_blake2b_init2(&lock_arg_state, 20*8, NULL, 0, (uint8_t*) blake2b_personalization, sizeof(blake2b_personalization)-1);
   cx_hash((cx_hash_t *) &lock_arg_state, 0, key_pair->public_key.W, key_pair->public_key.W_len, NULL, 0);
-  cx_hash((cx_hash_t *) &lock_arg_state, CX_LAST, NULL, 0, G.current_lock_arg, sizeof(G.current_lock_arg));
+  cx_hash((cx_hash_t *) &lock_arg_state, CX_LAST, NULL, 0, destination, sizeof(G.current_lock_arg));
   return 0;
 }
 
 #define P1_FIRST 0x00
 #define P1_NEXT 0x01
 #define P1_HASH_ONLY_NEXT 0x03 // You only need it once
+#define P1_CHANGE_PATH 0x10
+#define P1_CHANGE_PATH_CASE 0x11
```

Comment:
This is really just P1_CHANGE_PATH | P1_NEXT right

Reply:
Yes, but I was not able to rapidly verify that that's permissible in a case statement.